client-vms/src/vm/builder.ts

```diff
@@ -196,13 +196,20 @@ export class VmClientBuilder {
       }
     }
 
+    if (
+      !leaderClusterInfo.publicKeys ||
+      !leaderClusterInfo.publicKeys.authentication
+    )
+      throw new Error("Leader public key not in cluster details");
+
     const user_id = UserId.from(tokenAuthManager.publicKey);
     const payer = await new PaymentClientBuilder()
       .chainUrl(chainUrl)
       .paymentMode(supportedPaymentMode)
       .id(user_id)
       .signer(signer)
       .leader(leader.transport)
+      .leaderPublicKey(leaderClusterInfo.publicKeys?.authentication)
```

Comment:
nit: is this not type narrowed? eg it should be fine as  `.leaderPublicKey(leaderClusterInfo.publicKeys.authentication)`